### DISCUSSION TITLE
Log parsing error causes teler fail to analysis

### DISCUSSION THREAD

**Discussion by DummyKitty** [teler-sh/teler#128] [Q&A]
I tried to use the following log format, add $request_body to collect post data.
```nginx
    log_format  main  '$remote_addr $remote_user - [$time_iso8601] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$request_body" ';
```
teler's config file is:
```yaml
log_format: |
  $remote_addr $remote_user - [$time_local] "$request_method $request_uri $request_protocol" 
  $status $body_bytes_sent "$http_referer" "$http_user_agent" "$request_body" 
```

I use some data in the following format for testing
```nginx
192.168.68.129 - - [2021-03-11T01:31:26+00:00] "POST /index.php HTTP/1.1" 404 287 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" "id=%2e%2e%2f%2e%2e%2f%2e%2e%2f%2e%2e%2f%2e%2e%2f%7bFILE%7d&name=111asdasdaasd" 
```
But teler can not work properly.
![image-20210311093400339](https://user-images.githubusercontent.com/40905775/110724028-02266f80-8250-11eb-82ed-1ee0c6a1422f.png)

When I added a useless ending character "-" in nginx logfomat, change the teler's config file, and recreate testing data. 
```nginx
    log_format  main  '$remote_addr $remote_user - [$time_iso8601] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$request_body" -';
```
```yaml
log_format: |
  $remote_addr $remote_user - [$time_local] "$request_method $request_uri $request_protocol" 
  $status $body_bytes_sent "$http_referer" "$http_user_agent" "$request_body" $x   
```
```nginx
192.168.68.129 - - [2021-03-11T01:31:26+00:00] "POST /index.php HTTP/1.1" 404 287 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" "id=%2e%2e%2f%2e%2e%2f%2e%2e%2f%2e%2e%2f%2e%2e%2f%7bFILE%7d&name=111asdasdaasd" -
```
It worked.
![image-20210311093720964](https://user-images.githubusercontent.com/40905775/110724504-dd7ec780-8250-11eb-8a6f-c6e1fd58aac8.png)

I don't know why. I'm a freshman using Golang. Does this problem come from gonx package?

**Comment by dwisiswant0** [comment]
As we don't have clear documentation on how to define `log_format`, maybe this can help you: https://github.com/kitabisa/teler/discussions/86#discussioncomment-194019.
